src/test/java/io/strimzi/kafka/bridge/mqtt/MqttBridgetIT.java

```diff
@@ -0,0 +1,252 @@
+    private String mqttClientId;
+
+
+    /**
+     * Start the kafka Cluster
+     * Start the MQTT bridge before all tests
+     */
+    @BeforeAll
+    public static void beforeAll() throws InterruptedException {
+        final Map<String, Object> config = new HashMap<>();
+        String kafkaBootstrapServers = "localhost:9092";
+        try {
+            kafkaContainer = new StrimziKafkaContainer();
+            kafkaContainer.waitForRunning();
+            kafkaContainer.start();
+            kafkaBootstrapServers = kafkaContainer.getBootstrapServers();
+        } catch (Exception e) {
+            logger.error("Exception occurred", e);
+        }
+
+        // prepare the configuration for the bridge
+        config.put(KafkaConfig.KAFKA_CONFIG_PREFIX + ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers);
+        config.put(MqttConfig.MQTT_HOST, MQTT_SERVER_HOST);
+        config.put(MqttConfig.MQTT_PORT, MQTT_SERVER_PORT);
+        config.put(BridgeConfig.BRIDGE_ID, "my-bridge");
+
+        // instantiate the kafka consumer client
+        kafkaConsumerClient = new KafkaConsumer<>(
+                ImmutableMap.of(
+                        ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootstrapServers,
+                        ConsumerConfig.GROUP_ID_CONFIG, "gid-" + UUID.randomUUID(),
+                        ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, OffsetResetStrategy.EARLIEST.name().toLowerCase(Locale.ROOT)
+                ),
+                new StringDeserializer(),
+                new StringDeserializer()
+        );
+
+        // consumer client subscribes to the kafka topic
+        kafkaConsumerClient.subscribe(Collections.singletonList(KAFKA_TOPIC));
+
+        // start the bridge server in a separate thread
+        prepareServerThread = new Thread(() -> {
+            try {
+                EventLoopGroup bossGroup = new NioEventLoopGroup();
+                EventLoopGroup workerGroup = new NioEventLoopGroup();
+
+                BridgeConfig bridgeConfig = BridgeConfig.fromMap(config);
+
+                String mappingRulesPath = Objects.requireNonNull(MqttBridgetIT.class.getClassLoader().getResource("mapping-rules-regex.json")).getPath();
+                MappingRulesLoader.getInstance().init(mappingRulesPath);
+
+                mqttBridge = new MqttServer(bridgeConfig, bossGroup, workerGroup, ChannelOption.SO_KEEPALIVE);
+                mqttBridge.start();
+            } catch (InterruptedException interruptedException) {
+                // in this case, we are interrupting the thread, so we don't need to do anything
+                logger.info("Shutting down the server");
+            } catch (Exception e) {
+                logger.error("Exception occurred", e);
+            }
+        });
+        prepareServerThread.start();
+        Thread.sleep(5000);
+    }
+
+    /**
+     * Randomly create a new client id before each test
+     */
+    @BeforeEach
+    public void createClientId() {
```

Comment:
```suggestion
    public void generateMqttClientId() {
```

just to be clear it's about MQTT.
Also, do we need to story the MQTT client id in a local member of the class and having this as @BeforeEach? Can't make this private and call it in each test, or when needed?